aiohttp/web.py

```diff
@@ -277,6 +285,222 @@
+        finally:
+            if (
+                sigint_handler is not None
+                and signal.getsignal(signal.SIGINT) is sigint_handler
+            ):
+                signal.signal(signal.SIGINT, signal.default_int_handler)
+
+    def run_app(
+        self,
+        app: Union[Application, Awaitable[Application]],
+        *,
+        host: Optional[Union[str, HostSequence]] = None,
+        port: Optional[int] = None,
+        path: Union[PathLike, TypingIterable[PathLike], None] = None,
+        sock: Optional[Union[socket.socket, TypingIterable[socket.socket]]] = None,
+        shutdown_timeout: float = 60.0,
+        keepalive_timeout: float = 75.0,
+        ssl_context: Optional[SSLContext] = None,
+        print: Optional[Callable[..., None]] = print,
+        backlog: int = 128,
+        access_log_class: Type[AbstractAccessLogger] = AccessLogger,
+        access_log_format: str = AccessLogger.LOG_FORMAT,
+        access_log: Optional[logging.Logger] = access_logger,
+        handle_signals: bool = True,
+        reuse_address: Optional[bool] = None,
+        reuse_port: Optional[bool] = None,
+        handler_cancellation: bool = False,
+    ) -> None:
+        """Run an app locally"""
+        self._lazy_init()
+
+        self._loop.set_debug(self._debug)
+
+        if (
+            self._loop.get_debug()
+            and access_log
+            and access_log.name == "aiohttp.access"
+        ):
+            if access_log.level == logging.NOTSET:
+                access_log.setLevel(logging.DEBUG)
+            if not access_log.hasHandlers():
+                access_log.addHandler(logging.StreamHandler())
+
+        main_task = self._loop.create_task(
+            _run_app(
+                app,
+                host=host,
+                port=port,
+                path=path,
+                sock=sock,
+                shutdown_timeout=shutdown_timeout,
+                keepalive_timeout=keepalive_timeout,
+                ssl_context=ssl_context,
+                print=print,
+                backlog=backlog,
+                access_log_class=access_log_class,
+                access_log_format=access_log_format,
+                access_log=access_log,
+                handle_signals=handle_signals,
+                reuse_address=reuse_address,
+                reuse_port=reuse_port,
+                handler_cancellation=handler_cancellation,
+            )
+        )
+
+        try:
+            if self._set_event_loop:
+                asyncio.set_event_loop(self._loop)
+            self._loop.run_until_complete(main_task)
+        except (GracefulExit, KeyboardInterrupt):  # pragma: no cover
+            pass
+        finally:
+            _cancel_tasks({main_task}, self._loop)
+            _cancel_tasks(asyncio.all_tasks(self._loop), self._loop)
+            self._loop.run_until_complete(self._loop.shutdown_asyncgens())
+            self.close()
+            asyncio.set_event_loop(None)
```

Comment:
Already done in .close():
```suggestion
            _cancel_tasks({main_task}, self._loop)
            self.close()
```